crates/hotshot/src/traits/networking/libp2p_network.rs

```diff
@@ -298,7 +305,125 @@ where
     }
 }
 
-impl<M: NetworkMsg, K: SignatureKey + 'static> Libp2pNetwork<M, K> {
+/// Derive a Libp2p keypair from a given private key
+///
+/// # Errors
+/// If we are unable to derive a new `SecretKey` from the `blake3`-derived
+/// bytes.
+pub fn derive_libp2p_keypair<K: SignatureKey>(
+    private_key: &K::PrivateKey,
+) -> anyhow::Result<Keypair> {
+    // Derive a secondary key from our primary private key
+    let derived_key = blake3::derive_key("libp2p key", &(bincode::serialize(&private_key)?));
+    let derived_key = SecretKey::try_from_bytes(derived_key)?;
+
+    // Create an `ed25519` keypair from the derived key
+    Ok(ed25519::Keypair::from(derived_key).into())
```

Comment:
General question: why this secondary key pair is different from the SignatureKey scheme we used before? Do we want to add this as a new scheme for `SignatureKey`?